pkg/resource/replication_group/post_set_output.go

```diff
@@ -0,0 +1,70 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package replication_group
+
+import (
+	"context"
+
+	svcapitypes "github.com/aws-controllers-k8s/elasticache-controller/apis/v1alpha1"
+	"github.com/aws/aws-sdk-go/service/elasticache"
+)
+
+/*
+	To be called in sdkFind, this function updates the replication group's Spec fields with the latest observed state
+	This requires extra processing of the API response as well as additional API calls, and is necessary because
+	sdkFind does not update many of these Spec fields by default.
+*/
+func (rm *resourceManager) updateSpecFields(
+	ctx context.Context,
+	respRG *elasticache.ReplicationGroup,
+	ko *svcapitypes.ReplicationGroup,
```

Comment:
recommend just passing a `*resource` and do the `ko` unpacking inside the function. That way you don't need to import the `svcapitypes` package.